app/views/dishes/show.html.erb

```diff
@@ -0,0 +1,16 @@
+<h1>Dish</h1>
+
+<p>Name: <%= @dish.name %> </p>
+<p>Description: <%= @dish.description %> </p>
+<p>Ingredients: 
+  <%  @dish.ingredients.each do |ingredient| %>
+  <%= ingredient.name  %> <%= button_to "Remove #{ingredient.name}", "/chef/#{@chef.id}/dishes/#{@dish.id}/ingredients/#{ingredient.id}", method: :delete %> </b>
+  <% end %>
+</p>
+<p>Total Calories: <%= @dish.calories %></p>
+
+<%= form_with url: "/chef/#{@dish.chef.id}/dishes/#{@dish.id}", method: :patch, local: true, data: { turbo:false } do |form| %>
+<%= form.label :add_ingredient %>
+<%= form.text_field :add_ingredient %>
+<%= form.submit "Add" %>
+<% end %>
```

Comment:
While this form is what I would consider RESTful, it is not RESTful in what it's aiming to accomplish. The task we are given is to associate a new ingredient to a dish, this does not happen on the dish resource, this is a new relationship being CREATED on the DISH_INGREDIENTS table between a dish and an ingredient. We are not updating anything on the dish table itself, therefore a `PATCH` to ".../dishes/#{@dish.id}" is incorrect.

Therefore, your form should be making a `POST` request to `"/dishes/#{@dish.id}/ingredients"`.

```erb
<%= form_with url:"/dishes/#{@dish.id}/ingredients", method: :post, local: true, data: { turbo:false } do |form| %>
  <%= form.label :add_ingredient %>
  <%= form.number_field :add_ingredient %>
  <%= form.submit "Add" %>
<% end %>
```

Could I also suggest changing `:add_ingredient` to `:ingredient_id` 😅